### ISSUE TITLE
Restore `margin` parameter to `find_adversarial_example`

### ISSUE THREAD

**Issue by vtjeng** [vtjeng/MIPVerify.jl#160]
Adding a `margin` would enable users to ensure that the target category has logits strictly larger than any non-target category, accounting for solver tolerances. For more details, see https://github.com/vtjeng/MIPVerify.jl/issues/158#issuecomment-1997804778. 

## Sketch

- Ensure that `margin` is applied in all solve modes (`closest`; `worst`).
- Add tests that fail at HEAD and pass with the new behavior. It would probably be best to use a simple NN.

NOTE: This would partially undo https://github.com/vtjeng/MIPVerify.jl/pull/65.
